mr/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -0,0 +1,571 @@
+   */
+  public static ConfigBuilder configure(Job job) {
+    job.setInputFormatClass(IcebergInputFormat.class);
+    return new ConfigBuilder(job.getConfiguration());
+  }
+
+  public static class ConfigBuilder {
+    private final Configuration conf;
+
+    public ConfigBuilder(Configuration conf) {
+      this.conf = conf;
+    }
+
+    public ConfigBuilder readFrom(String path) {
+      conf.set(TABLE_PATH, path);
+      Table table = findTable(conf);
+      conf.set(TABLE_SCHEMA, SchemaParser.toJson(table.schema()));
+      return this;
+    }
+
+    public ConfigBuilder filter(Expression expression) {
+      conf.set(FILTER_EXPRESSION, SerializationUtil.serializeToBase64(expression));
+      return this;
+    }
+
+    public ConfigBuilder project(Schema schema) {
+      conf.set(READ_SCHEMA, SchemaParser.toJson(schema));
+      return this;
+    }
+
+    public ConfigBuilder reuseContainers(boolean reuse) {
+      conf.setBoolean(REUSE_CONTAINERS, reuse);
+      return this;
+    }
+
+    public ConfigBuilder caseSensitive(boolean caseSensitive) {
+      conf.setBoolean(CASE_SENSITIVE, caseSensitive);
+      return this;
+    }
+
+    public ConfigBuilder snapshotId(long snapshotId) {
+      conf.setLong(SNAPSHOT_ID, snapshotId);
+      return this;
+    }
+
+    public ConfigBuilder asOfTime(long asOfTime) {
+      conf.setLong(AS_OF_TIMESTAMP, asOfTime);
+      return this;
+    }
+
+    public ConfigBuilder splitSize(long splitSize) {
+      conf.setLong(SPLIT_SIZE, splitSize);
+      return this;
+    }
+
+    /**
+     * If this API is called. The input splits
+     * constructed will have host location information
+     */
+    public ConfigBuilder preferLocality() {
+      conf.setBoolean(LOCALITY, true);
+      return this;
+    }
+
+    public ConfigBuilder catalogFunc(Class<? extends Function<Configuration, Catalog>> catalogFuncClass) {
+      Preconditions.checkState(
+          conf.get(TABLE_PATH) == null,
+          "Please provide custom catalog before specifying the table to read from");
+      conf.setClass(CATALOG, catalogFuncClass, Function.class);
+      return this;
+    }
+
+    public ConfigBuilder useHiveRows() {
+      conf.set(IN_MEMORY_DATA_MODEL, InMemoryDataModel.HIVE.name());
+      return this;
+    }
+
+    public ConfigBuilder usePigTuples() {
+      conf.set(IN_MEMORY_DATA_MODEL, InMemoryDataModel.PIG.name());
+      return this;
+    }
+
+    /**
+     * Compute platforms pass down filters to data sources.
+     * If the data source cannot apply some filters, or only
+     * partially applies the filter, it will return the
+     * residual filter back. If the platform
+     * can correctly apply the residual filters, then it
+     * should call this api. Otherwise the current
+     * api will throw an exception if the passed in
+     * filter is not completely satisfied. Note: This
+     * does not apply to standalone MR application
+     */
+    public ConfigBuilder platformAppliesFilterResiduals() {
```

Comment:
I find this a little hard to reason about and I think a different name could make it more straight-forward. Rather than configuring "someone will do task" it is easier for the caller and this code to phrase it as "do task" or "skip task". That also avoids the need to refer to the "platform" because users won't know what that is and we have to explain it.

What about changing this to `skipResidualFiltering()`? That is clear that the `InputFormat` is not responsible for filtering, but doesn't define who or what is responsible. It is also clearly dangerous, even if you don't know what a residual is.

Reply:
fixed